lock-api/src/test/java/com/palantir/lock/client/MultiClientTransactionStarterTest.java

```diff
@@ -135,6 +137,7 @@ public void servesRequestsAsSoonAsResponseIsReceived() {
         LockCleanupService relevantLockCleanupService = LOCK_CLEANUP_SERVICE_MAP.get(namespace);
         verify(relevantLockCleanupService).refreshLockLeases(any());
         verify(relevantLockCleanupService).unlock(any());
+        verify(NAMESPACE_CACHE_MAP.get(namespace), times(4)).removeTransactionStateFromCache(anyLong());
```

Comment:
These numbers are magical to me; @sudiksha27 mind giving me some context so I can at least comment on why this is 4 (or 5 below)?

Reply:
Right, so, for one namespace, we accumulate all requests for it in ResponseHandler#pendingFutures i.e. this queue is like [1, 25] for this test. Then we try to fetch as many startTs as possible from the server. For each batch of startTxnReponses, we try to server as many futures as we can. In one go our limit is 5 for these tests. So basically for one namespace i can fetch 5 responses.
If you see ResponseHandler#serveRequests, you will find, we greedily server requests from the pool of responses and the rest of the responses are transient i.e. they may be served if we have enough responses from future server calls. 
In this test, we can serve 1st request that needs 1 response. The next req needs 25 so we cannot serve it and therefore, there are 4 requests that need clearing up (server is configured to throw after one call)